core/src/main/kotlin/entity/Asset.kt

```diff
@@ -0,0 +1,90 @@
+package dev.kord.core.entity
+
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.KordObject
+import dev.kord.rest.Image
+import dev.kord.rest.route.CdnUrl
+import dev.kord.rest.route.DiscordCdn
+
+public sealed class Asset(
+    public val format: Image.Format,
```

Comment:
why is the default format png when it can't be animated and webp if the hash says it's not animated?

i think there should be other constructors instead:
the primary constructor as it is right now but with a default value for format (png or webp, whichever is the right default) and another constructors like this:
```kotlin
constructor(animated: Boolean, cdnUrl: CdnUrl, kord: Kord) :
    this(format = if (anmiated) GIF else <default format>, cdnUrl, kord)
```